### PR TITLE
DX: use only PHP modules that are required

### PR DESCRIPTION
**Pull request by kubawerlos**
Goal: performance? Smaller resource usage?

Initial situation (see logs from 1st commit):
```bash
php -m | wc -l && php -m
88
[PHP Modules]
amqp
apcu
ast
bcmath
bz2
calendar
Core
ctype
curl
date
dba
dom
ds
enchant
exif
FFI
fileinfo
filter
ftp
gd
gettext
gmp
hash
iconv
igbinary
imagick
imap
intl
json
ldap
libxml
mbstring
memcache
memcached
mongodb
msgpack
mysqli
mysqlnd
odbc
openssl
pcntl
pcre
PDO
pdo_dblib
PDO_Firebird
pdo_mysql
PDO_ODBC
pdo_pgsql
pdo_sqlite
pdo_sqlsrv
pgsql
Phar
posix
pspell
random
readline
redis
Reflection
session
shmop
SimpleXML
snmp
soap
sockets
sodium
SPL
sqlite3
sqlsrv
standard
sysvmsg
sysvsem
sysvshm
tidy
tokenizer
xml
xmlreader
xmlwriter
xsl
yaml
Zend OPcache
zip
zlib
zmq

[Zend Modules]
Zend OPcache
```

After changes in this PR (see commit "Add zip"):
```bash
php -m | wc -l && php -m
29
[PHP Modules]
Core
curl
date
dom
filter
hash
json
libxml
mbstring
openssl
pcntl
pcre
Phar
random
Reflection
session
SimpleXML
sodium
SPL
standard
tokenizer
xml
xmlwriter
zip
zlib

[Zend Modules]
```

### CODE REVIEW

**Review by Wirone — APPROVED** [review]
I don't know whether we should merge it. Comparing CI workflows, the "Setup PHP" step takes similar time before and after change, so there's no gain in terms of running jobs.

The good part is that we have explicit list of extensions and we don't bloat runtime with them. If there is any code that depends on the list of installed extensions, it now has less to process. But it is micro-optimisation 😅.

Im am not against it, but at the same time I am not in favor that much. I am going to give approval and you can decide @kubawerlos.

Or maybe @keradus can look at it too.

**Comment by kubawerlos** [comment]
> Comparing CI workflows, the "Setup PHP" step takes similar time before and after change, so there's no gain in terms of running jobs.

@Wirone TBH I didn't even look at the times of separate steps, I've compared times of jobs - this PR:
image

with times of [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6958) PR:
image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -94,6 +94,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php-version }}
+          extensions: none, dom, json, opcache, simplexml, tokenizer, xml, xmlwriter, zip
```
[the remark this answers is not in the document]
Comment (kubawerlos) [reply .github/workflows/ci.yml:97]: Yes, that is the way this GA needs to be configured.

**Comment by Wirone** [comment]
Those times are random and undeterministic 😅. Some are better, some are worse 🤷‍♂️. It's OK for me both ways, I don't have much experience with that action so I can't tell which approach is better. I can ask somewhere 😉.

**Comment by kubawerlos** [comment]
They are not that random, job "CI / PHP 8.2 with calculating code coverage (pull_request)" (usually the slowest one on linux) is 23 minutes [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6961), 18 [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6960) and 16 [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6952) - 3 recent PRs with passing build - so nowhere near 11 minutes as in this PR.

**Comment by Wirone** [comment]
Thank you @kubawerlos 🍻

**Pull request merged**